components/support/rc_crypto/nss/src/pkixc.rs

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::error::*;
+use crate::util::ensure_nss_initialized;
+
+use nss_sys::PRErrorCode;
+
+// NSS error codes.
+// https://searchfox.org/mozilla-central/source/security/nss/lib/util/secerr.h#29
+// https://searchfox.org/mozilla-central/source/security/nss/lib/mozpkix/include/pkix/Result.h
+const SEC_ERROR_UNKNOWN_ISSUER: i32 = -8179; // -8192 + 13
+const SEC_ERROR_EXPIRED_ISSUER_CERTIFICATE: i32 = -8162; // -8192 + 30
+const SEC_ERROR_SUBJECT_MISMATCH: i32 = -12276; // ???
+const SEC_ERROR_EXPIRED_CERTIFICATE: i32 = -16378; // ???
+
+const ROOT_HASH_LENGTH: usize = 32;
+
+pub fn verify_code_signing_certificate_chain(
+    certificates: Vec<&[u8]>,
+    seconds_since_epoch: u64,
+    root_sha256_hash: &[u8],
+    hostname: &str,
+) -> Result<()> {
+    ensure_nss_initialized();
+
+    let mut cert_lens: Vec<u16> = vec![];
+    for certificate in &certificates {
+        cert_lens.push(certificate.len() as u16);
```

Comment:
Is it worth erroring out here if the length is too large to fit in a `u16`? IIUC Rust will just truncate the bits when casting to a smaller integer size, which will always produce a smaller length, which will not risk memory unsafety...so I guess it's fine? May be worth a comment to that effect at least.